rust/noosphere-core/src/data/address.rs

```diff
@@ -65,13 +74,88 @@ pub struct LinkRecord(Jwt);
 impl LinkRecord {
     /// Parse the wrapped [Jwt] as a [Ucan] and looks for the referenced pointer
     /// to some data in IPFS (via a [Cid] in the `fct` field).
-    pub async fn dereference(&self) -> Option<Cid> {
-        let token = &self.0;
-        let ucan = match Ucan::try_from(token.to_string()) {
-            Ok(ucan) => ucan,
-            _ => return None,
+    pub fn dereference(&self) -> Option<Cid> {
+        match Ucan::try_from(self) {
+            Ok(ucan) => LinkRecord::link(&ucan),
+            _ => None,
+        }
+    }
+
+    /// Validates the [Ucan] token as a [LinkRecord], ensuring that
+    /// the sphere's owner authorized the publishing of a new
+    /// content address. Returns an `Err` if validation fails.
```

Comment:
It would probably be good to note the context-specific rules for expiry validation here in this comment.